bench/src/main/scala/cats/bench/TraverseBench.scala

```diff
@@ -21,155 +21,182 @@
 
 package cats.bench
 
-import cats.{Always, Applicative, Eval, Traverse}
-import cats.instances.either._
-import cats.instances.list._
-import cats.instances.vector._
-import org.openjdk.jmh.annotations.{Benchmark, Scope, State}
+import cats.{Eval, Traverse, TraverseFilter}
+import org.openjdk.jmh.annotations.{Benchmark, Param, Scope, Setup, State}
+import org.openjdk.jmh.infra.Blackhole
 
 @State(Scope.Benchmark)
-class TraverseListBench {
-  val instance: Traverse[List] = Traverse[List]
-  val f: Int => Either[Int, Int] = Right(_)
-
-  val xs1: List[Int] = (1 to 10).toList
-  val xs2: List[Int] = (1 to 100).toList
-  val xs3: List[Int] = (1 to 1000).toList
-  val xs4: List[Int] = (1 to 10000).toList
-
-  def traverseCats[G[_], A, B](fa: List[A])(f: A => G[B])(implicit G: Applicative[G]): G[List[B]] =
-    instance
-      .foldRight[A, G[List[B]]](fa, Always(G.pure(List.empty))) { (a, lglb) =>
-        G.map2Eval(f(a), lglb)(_ :: _)
+class TraverseBench {
+  val listT: Traverse[List] = Traverse[List]
+  val listTFilter: TraverseFilter[List] = TraverseFilter[List]
+
+  val vectorT: Traverse[Vector] = Traverse[Vector]
+  val vectorTFilter: TraverseFilter[Vector] = TraverseFilter[Vector]
+
+  // the unit of CPU work per iteration
+  private[this] val Work: Long = 10
```

Comment:
In terms of JMH, we are going to burn CPU time per _invocation_ here. Isn't it?

Reply:
Yep! Specifically, per iteration of the `traverse` itself.